FE/src/components/NewRecipe.jsx

```diff
@@ -1,8 +1,165 @@
+
+   function handleIngredientAdd() {
+    setIngredientList([...ingredientList, {ingredient: ''}]);
+   };
+
+   function handleInstructionAdd() {
+    setInstructionList([...instructionList, {instruction: ''}]);
+   };
+
+   function handleNoteAdd() {
+    setNotesList([...notesList, {note: ''}]);
+   };
+
+   function handleIngredientDelete(index) {
+    const list = [...ingredientList];
+    list.splice(index, 1);
+    setIngredientList(list);
+   };
+
+   function handleInstructionDelete(index) {
+    const list = [...instructionList];
+    list.splice(index, 1);
+    setInstructionList(list);
+   };
+
+   function handleNoteDelete(index) {
+    const list = [...notesList];
+    list.splice(index, 1);
+    setNotesList(list);
+   };
+
+   function handleIngredientChange(e, index) {
+    e.preventDefault();
+    const {name, value} = e.target;
+    const list = [...ingredientList];
+    list[index][name] = value;
+    setIngredientList(list);
+   };
+
+   function handleInstructionChange(e, index) {
+    e.preventDefault();
+    const {name, value} = e.target;
+    const list = [...instructionList];
+    list[index][name] = value;
+    setInstructionList(list);
+   };
+
+   function handleNoteChange(e, index) {
+    e.preventDefault();
+    const {name, value} = e.target;
+    const list = [...notesList];
+    list[index][name] = value;
+    setNotesList(list);
+   };
+
+   function rearrange (input, field) {
+    let output = [];
+    for (let i=0; i < input.length; i++) {
+        output.push(input[i][field]);
+    }
+    return output;
+   }
+
+   function handleSubmit(e) {
+        e.preventDefault();
+
+        let ingredArray = rearrange(ingredientList, 'ingredient');
+        let instructArray = rearrange(instructionList, 'instruction');
+        let noteArray = rearrange(notesList, 'note');
+
+
+        let data = {
+            title: title,
+            estTime: estTime,
+            ingredients: ingredArray,
+            procedure: instructArray,
+            image: image,
+            notes: noteArray,
+            tags: tags
+        };
+
+        console.log(data)
+    }
+
+   return (
+       <div>
+           <form onSubmit={handleSubmit}>
+               <label>Title: </label>
+               <input type="text" id="title" name="title" value={title} onChange={(e) => {
+                setTitle(e.target.value);
+               }}/>
+
+               <label>Estimated Time: </label>
+               <input list="times" id="estTime" name="estTime" value={estTime} onChange={(e) => {
+                setEstTime(e.target.value);
+               }}/>
+                   <datalist id="times">
+                       <option value="15 min"/>
+                       <option value="30 min"/>
+                       <option value="45 min"/>
+                       <option value="60 min"/>
+                       <option value="75 min"/>
+                       <option value="90 min"/>
+                   </datalist>
+
```

Comment:
For easier viewing on web maybe add a <br/> to put the first ingredient on a new line

Reply:
Maybe put <br/> between each element?